include/oxen/quic/ip.hpp

```diff
@@ -46,13 +55,21 @@ namespace oxen::quic
         ipv4 base;
         uint8_t mask;
 
-        const std::string to_string() const { return base.to_base(mask).to_string(); }
+        constexpr ipv4 max_ip() const
+        {
+            auto b = base.to_base(mask);
 
-        constexpr bool operator==(const ipv4_net& a) const { return base == a.base && mask == a.mask; }
+            if (mask < 32)
+                b.addr |= (1 << (32 - mask)) - 1;
```

Comment:
```suggestion
                b.addr |= (uint32_t{1} << (32 - mask)) - 1;
```
to match the explicit type in the ipv6 version.  I think this is implicitly okay as is (because `int << uint8_t` gets promoted to `unsigned int << unsigned int`), but the explicit uint32_t there makes it more obviously correct.